### PR TITLE
CBO-789-Accessibility-updates-to-remaining-views

### PR DESCRIPTION
**Pull request by naseberry**
#### What
Accessibility and coding standard updates to views not covered in previously.

#### Ticket
[CBO-789](https://dsdmoj.atlassian.net/browse/CBO-789)

#### Why
To resolve accessibility issues.

#### How
- Ensure elements have sufficient color contrast
- Improved link descriptions
- Ensure table headers in a data table refers to data cells
- Improve coding standard

### CODE REVIEW

File: app/views/external_users/claims/misc_fees/litigators/_misc_fee_fields.html.haml [hunk 1]
```diff
@@ -26,6 +26,7 @@
 
       = validation_error_message(@error_presenter, "misc_fee_#{@misc_fee_count+1}_fee_type")
 
+  %a{ id: "misc_fee_#{@misc_fee_count}_amount" }
```
Comment (jsugarman) [inline app/views/external_users/claims/misc_fees/litigators/_misc_fee_fields.html.haml:29]: Am not sure, but does the adp_text_field not create an anchor like this?

File: app/views/layouts/_footer.html.haml [hunk 2]
```diff
@@ -1,13 +1,11 @@
 %ul
   %li
-    %a{:href => "http://www.gov.uk/help/cookies"} Cookies
+    = link_to t('cookies'), 'http://www.gov.uk/help/cookies'
```
Comment (jsugarman) [inline app/views/layouts/_footer.html.haml:3]: 🎖

File: app/views/layouts/application.html.haml [hunk 3]
```diff
@@ -28,13 +28,14 @@
...
-      %iframe{ :src => "https://www.googletagmanager.com/ns.html?id=#{ENV['GTM_TRACKED_ID']}", :style => "display:none;visibility:hidden",:height => "0", :width => "0"}
+      %iframe{ src: "https://www.googletagmanager.com/ns.html?id=#{ENV['GTM_TRACKED_ID']}", style: 'display:none;visibility:hidden', height: '0', width: '0' }
 
 - content_for :proposition_header do
   .header-proposition
     .content
-      %a.js-header-toggle.menu{:href => "#proposition-links"}
-        = "Menu"
+      - unless current_user.nil?
+        = link_to 'Menu', '#proposition-links', class: 'js-header-toggle menu'
```
Comment (jsugarman) [inline app/views/layouts/application.html.haml:37]: 🎖

File: app/views/pages/api_release_notes.html.haml [hunk 4]
```diff
@@ -979,36 +979,34 @@
...
 
   %hr
-  %h1 October 2016
+  %h2 October 2016
   %ul.list-bullet
     %li
       Users of the api-sandbox were having difficulties because the ids of the static data records (expense types, fee types etc) were
       liable to change every time we regenerated the database.  Over the weekend of 8th-9th October, the database was recreated and the ids
       of these records fixed to match those on the live database.  These ids are now static and will not change in future.
-    %p
```
Comment (jsugarman) [inline app/views/pages/api_release_notes.html.haml:1011]: **Not a blocker** These paras gave some extra spacing
**old**
Screen Shot 2019-10-17 at 13 55 53

**new** 
Screen Shot 2019-10-17 at 13 56 54

File: app/views/pages/api_release_notes.html.haml [hunk 5]
```diff
@@ -511,133 +509,135 @@
...
+            ** see not below
+  %ul.list-bullet
     %li
       "Cracked case discontinued" fixed fee type and its uplift has been deprecated.
       %br/
       %br/
       The following fixed fee type has been deprecated and should not be used.
       %br/
       %br/
-      %table
```
Comment (jsugarman) [inline app/views/pages/api_release_notes.html.haml]: **Not a blocker** This nesting kept the table within the `%li`

File: app/views/pages/api_landing.html.haml [hunk 6]
```diff
@@ -77,8 +77,8 @@
             %li
               disbursement types
 
-            %p
-              Each endpoint returns a set of JSON formatted objects, each with an id, which you can use when submitting a claim and associated fees and expenses.
+          %p
```
Comment (jsugarman) [inline app/views/pages/api_landing.html.haml]: **Not a blocker** needs indenting to be with the `%li`

File: app/views/pages/api_release_notes.html.haml [hunk 7]
```diff
@@ -126,21 +124,21 @@
...
-    %br
-    .panel
-      'Other offences' in bands 1.1, 1.2, 1.3, 1.4, 2.1, 2.2, 3.1, 3.2, 3.3, 3.4, 3.5, 4.1, 4.2, 4.3, 5.1, 5.2, 5.3, 6.1, 6.2, 6.3, 6.4, 6.5, 7.1, 7.2, 7.3, 8.1, 9.1, 9.2, 9.3, 9.4, 9.5, 9.6, 9.7, 10.1, 11.1, 11.2, 12.1, 12.2, 12.3, 13.1, 14.1, 15.1, 15.2, 15.3, 16.1, 16.2, 16.3, 17.1
-    %br/
-    You can query all offences on the
-    %a{ href: '/api/documentation?v=1#!/api/getApiOffences', target: '_blank' } api/offences
-    endpoint - note that you will need to provide a
-    %code.code='rep_order_date'
-    parameter that falls within the scheme 10/11 scheme start and end date - start dates are 2018-04-01 and 2018-12-31 respectively
+      %br/
```
Comment (jsugarman) [inline app/views/pages/api_release_notes.html.haml:127]: **Not a blocker.** This indent removes a line break. could just add another `%br`

**old**
Screen Shot 2019-10-17 at 14 07 30

**new
Screen Shot 2019-10-17 at 14 08 02
**

**Review by jsugarman — APPROVED** [review]
👍 Looks good, just some minor/non-blocking line break issues in the api documentation pages

File: app/views/external_users/claims/misc_fees/litigators/_misc_fee_fields.html.haml (hunk 1 above)
Comment (naseberry) [reply app/views/external_users/claims/misc_fees/litigators/_misc_fee_fields.html.haml:29]: This was removed in an earlier sprint as it caused a "duplicate id" issue.

**Pull request merged**